setup.py

```diff
@@ -33,27 +33,21 @@
 #
 
 # import only from standard library so dependencies can be installed
-try:
-    # use setuptools if we can
-    from setuptools import setup, Command
-    from setuptools.command.build_ext import build_ext
-    using_setuptools = True
-except ImportError:
-    from distutils.core import setup, Command
-    from distutils.command.build_ext import build_ext
-    using_setuptools = False
-
-from distutils import log
-from distutils.ccompiler import get_default_compiler, new_compiler
+from setuptools import setup, Command
+from setuptools.command.build_ext import build_ext
+
 from distutils.errors import CompileError, LinkError
```

Comment:
Let's leave it to a future PR, but we need to eventually move away from `distutils` going forward. It is deprecated in Python 3.10 and will be removed in Python 3.12 (if I recall correctly).

Reply:
The documentation for `setuptools` is only marginally better than that of `distutils`. I didn't find an easy guide to update this to a future-compatible `setup.py`. I hope that once Python 3.12 approaches, the documentation on how to switch away from `distutils` will be a lot better.